pkg/lib/verifier/rule_builder_util.go

```diff
@@ -17,112 +18,174 @@ import (
 	"github.com/pkg/errors"
 )
 
-func getPcrMatchesConstantRules(pcrs []types.PcrIndex, flavor *hvs.Flavor, marker common.FlavorPart) ([]rules.Rule, error) {
+//getPcrMatchesConstantRules method will create PcrMatchesConstantRule and return the rule
+//return nil if error occurs
+func getPcrMatchesConstantRules(pcrs []types.PcrIndex, flavor *hvs.Flavor, pcrLogData *types.PCRS, marker common.FlavorPart) ([]rules.Rule, error) {
 
 	var results []rules.Rule
+	var rule rules.Rule
+	var err error
 
-	// iterate over the banks, collecting the values for each supplied index
-	// and create PcrMatchesConstant rules.
-	for bank, pcrMap := range flavor.Pcrs {
-		for _, index := range pcrs {
-			if expectedPcrEx, ok := pcrMap[index.String()]; ok {
-				expectedPcr, _ := rules.FlavorPcr2ManifestPcr(&expectedPcrEx, types.SHAAlgorithm(bank), index)
+	if pcrLogData != nil {
 
-				rule, err := rules.NewPcrMatchesConstant(expectedPcr, marker)
-				if err != nil {
-					return nil, errors.Wrapf(err, "An error occurred creating a PcrMatchesConstant rule for bank '%s', index '%s'", bank, index)
+		rule, err = rules.NewPcrMatchesConstant(nil, pcrLogData, marker)
+		if err != nil {
+			return nil, errors.Wrapf(err, "An error occurred creating a PcrMatchesConstant rule ")
+		}
+	} else {
+		// iterate over the banks, collecting the values for each supplied index
+		// and create PcrMatchesConstant rules.
+		for bank, pcrMap := range flavor.Pcrs {
+			for _, index := range pcrs {
+				if expectedPcrEx, ok := pcrMap[index.String()]; ok {
+					expectedPcr, _ := rules.FlavorPcr2ManifestPcr(&expectedPcrEx, types.SHAAlgorithm(bank), index)
+
+					rule, err := rules.NewPcrMatchesConstant(expectedPcr, nil, marker)
+					if err != nil {
+						return nil, errors.Wrapf(err, "An error occurred creating a PcrMatchesConstant rule for bank '%s', index '%s'", bank, index)
+					}
+
+					results = append(results, rule)
 				}
-
-				results = append(results, rule)
 			}
 		}
 	}
+	results = append(results, rule)
 
 	return results, nil
 }
 
-func getPcrEventLogEqualsExcludingRules(pcrs []types.PcrIndex, flavor *hvs.Flavor, marker common.FlavorPart) ([]rules.Rule, error) {
+//getPcrEventLogEqualsRules method will create PcrEventLogEqualsRule and return the rule
+//return nil if error occurs
+func getPcrEventLogEqualsRules(pcrs []types.PcrIndex, pcrLogData *types.PCRS, flavor *hvs.Flavor, marker common.FlavorPart) ([]rules.Rule, error) {
 
 	var results []rules.Rule
```

Comment:
Using variable names as 'rules' instead of 'results' is showing error because we are using package name called 'rules'. Hence it is conflicting. Can we use some other name like 'pcrRules'?

Reply:
yes, Sure